bittensor/axon.py

```diff
@@ -888,25 +890,38 @@ async def default_verify(self, synapse: bittensor.Synapse):
             # Build the unique endpoint key.
             endpoint_key = f"{synapse.dendrite.hotkey}:{synapse.dendrite.uuid}"
 
-            # Check the nonce from the endpoint key with 4 second delta
-            allowedDelta = 4000000000
-
             # Requests must have nonces to be safe from replays
             if synapse.dendrite.nonce is None:
                 raise Exception("Missing Nonce")
 
             # If we don't have a nonce stored, ensure that the nonce falls within
             # a reasonable delta.
+
+            # Updated nonce using NTP implementated at v7.2
             if (
-                self.nonces.get(endpoint_key) is None
-                and synapse.dendrite.nonce <= time.time_ns() - allowedDelta
-            ):
-                raise Exception("Nonce is too old")
-            if (
-                self.nonces.get(endpoint_key) is not None
-                and synapse.dendrite.nonce <= self.nonces[endpoint_key]
+                synapse.dendrite.version is not None
+                and synapse.dendrite.version >= V_7_2_0
             ):
-                raise Exception("Nonce is too old")
+                # If we don't have a nonce stored, ensure that the nonce falls within
+                # a reasonable delta.
+                current_time = BittensorNTPClient.get_current_ntp_time()
+                if (
+                    self.nonces.get(endpoint_key) is None
+                    and synapse.dendrite.nonce <= current_time - ALLOWED_DELTA
+                ):
+                    raise Exception("Nonce is too old")
+                if (
+                    self.nonces.get(endpoint_key) is not None
+                    and synapse.dendrite.nonce <= self.nonces[endpoint_key]
+                ):
+                    raise Exception("Nonce is too old")
+            else:
+                if (
+                    endpoint_key in self.nonces.keys()
+                    and self.nonces[endpoint_key] is not None
+                    and synapse.dendrite.nonce <= self.nonces[endpoint_key]
```

Comment:
there is bug in here, since if we sent N requests at the same time to the same axon, they can very well return in different order, but I guess it can be resolved another time as it is not a common pattern to do that.